DigitalLearningSolutions.Web/Controllers/RegisterController.cs

```diff
@@ -40,66 +42,62 @@ CustomPromptHelper customPromptHelper
             this.customPromptHelper = customPromptHelper;
         }
 
-        public IActionResult Index()
+        public IActionResult Index(int? centreId)
         {
             if (User.Identity.IsAuthenticated)
             {
                 return RedirectToAction("Index", "Home");
             }
 
-            var delegateRegistrationData = TempData.Peek<DelegateRegistrationData>();
-
-            if (delegateRegistrationData == null || !Request.Cookies.ContainsKey(CookieName))
+            if (!CheckCentreIdValid(centreId))
             {
-                delegateRegistrationData = new DelegateRegistrationData();
-                var id = delegateRegistrationData.Id;
-
-                Response.Cookies.Append(
-                    CookieName,
-                    id.ToString(),
-                    new CookieOptions
-                    {
-                        Expires = DateTimeOffset.UtcNow.AddDays(30)
-                    }
-                );
-                TempData.Set(delegateRegistrationData);
+                return NotFound();
             }
 
-            ViewBag.CentreOptions = SelectListHelper.MapOptionsToSelectListItems(
-                centresDataService.GetActiveCentresAlphabetical(),
-                delegateRegistrationData.RegisterViewModel.Centre
-            );
+            var delegateRegistrationData = CreateDelegateRegistrationData(centreId);
+            delegateRegistrationData.IsCentreSpecificRegistration = centreId.HasValue;
+            delegateRegistrationData.Centre = centreId;

```

Comment:
Doesn't this get set in the "CreateDelegateRegistrationData"? Also, if we named it "SetDelegateRegistrationData", we could move the TempData.Set into the method